config/legacyconfig/migrate.go

```diff
@@ -1048,6 +1059,69 @@ func migrateFlag(key, value string, newConfig *config.Config) error {
 	return nil
 }
 
+// migrateExportKubeConfigFlags migrates 'kube-config-context-name', 'out-kube-config-server', 'out-kube-config-secret'
+// and 'out-kube-config-secret-namespace' flags.
+//
+// The migration is done in the following way:
+//   - 'kube-config-context-name' and 'out-kube-config-server', if set, are always migrated into ExportKubeConfig.Context
+//     and ExportKubeConfig.Server config properties, respectively.
+//   - 'out-kube-config-secret' and 'out-kube-config-secret-namespace', if set, are migrated as the first additional
+//     secret in the new ExportKubeConfig.AdditionalSecrets config. When this first additional kubeconfig secret config
+//     is created, it also gets Context and Server properties from 'kube-config-context-name' and 'out-kube-config-server'.
+//     This is done in order to preserve the previous behavior where the additional kubeconfig Secret, specified in
+//     ExportKubeConfig.Secret, used context and server values from ExportKubeConfig.Context and ExportKubeConfig.Server.
+func migrateExportKubeConfigFlags(exportKubeConfigFlags map[string]string, exportKubeConfig *config.ExportKubeConfig) error {
+	if exportKubeConfig == nil {
+		return errors.New("exportKubeConfig is not set")
+	}
+
+	ensureAdditionalSecretIsSet := func() {
+		if len(exportKubeConfig.AdditionalSecrets) == 0 {
+			exportKubeConfig.AdditionalSecrets = []config.ExportKubeConfigAdditionalSecretReference{{}}
+		}
+	}
+
+	// We will set the additional secret only if the additional secret name and/or namespace flags are set.
+	var setAdditionalSecret bool
+	if secretName, ok := exportKubeConfigFlags[kubeConfigAdditionalSecretNameFlag]; ok {
+		if secretName == "" {
+			return fmt.Errorf("%s flag value is missing", kubeConfigAdditionalSecretNameFlag)
+		}
+		ensureAdditionalSecretIsSet()
+		exportKubeConfig.AdditionalSecrets[0].Name = secretName
+		setAdditionalSecret = true
+	}
+	if secretNamespace, ok := exportKubeConfigFlags[kubeConfigAdditionalSecretNamespaceFlag]; ok {
+		if secretNamespace == "" {
+			return fmt.Errorf("%s flag value is missing", kubeConfigAdditionalSecretNamespaceFlag)
+		}
+		ensureAdditionalSecretIsSet()
+		exportKubeConfig.AdditionalSecrets[0].Namespace = secretNamespace
+		setAdditionalSecret = true
+	}
+
+	if context, ok := exportKubeConfigFlags[kubeConfigContextFlag]; ok {
+		if context == "" {
+			return fmt.Errorf("%s flag value is missing", kubeConfigContextFlag)
+		}
+		exportKubeConfig.Context = context
+		if setAdditionalSecret {
+			exportKubeConfig.AdditionalSecrets[0].Context = context
+		}
+	}
+	if server, ok := exportKubeConfigFlags[kubeConfigServerFlag]; ok {
+		if server == "" {
+			return fmt.Errorf("%s flag value is missing", kubeConfigServerFlag)
+		}
+		exportKubeConfig.Server = server
+		if setAdditionalSecret {
+			exportKubeConfig.AdditionalSecrets[0].Server = server
+		}
+	}
+
+	return nil
+}
+
```

Comment:
Since there is an existing [test](https://github.com/loft-sh/vcluster/blob/a4bdf0beed4e5f85f1e07050cb288fb082f614f3/config/legacyconfig/migrate_test.go) for `MigrateLegacyConfig`, could you add some test cases there to ensure the conversion is as expected?

Reply:
Good point, will do that.

Reply:
Added new test cases that include extraArgs being migrated into exportKubeConfig.